space_packet_parser/xtcedef.py

```diff
@@ -984,13 +1004,33 @@ def read(self, format_string: str, update_position: bool = True) -> Union[int, f
         if update_position:
             self.pos += nbits
 
-        if name == "uint":
+        # TODO: This logic is messy. When we tackle getting away from the bitstring format strings,
+        #  refactor this to be more concise
```

Comment:
Yeah, agreed. I'll try and profile it later if you haven't already, but I'd like to make sure that this isn't slowing things up _too_ much with all these added cases to check.

Reply:
On my local test case (IDEX packet definition), I went from 6.6 to 6.8 seconds. So not miserable, but it did show up. I think this is acceptable as-is to keep moving forward.